evm/src/trace/decoder.rs

```diff
@@ -72,6 +78,8 @@ pub struct CallTraceDecoder {
     pub events: BTreeMap<(H256, usize), Vec<Event>>,
     /// All known errors
     pub errors: Abi,
+    /// A signature identifier for events and functions.
+    pub signature_identifier: Option<Arc<RwLock<SignaturesIdentifier>>>,
```

Comment:
why do we need to guard this?

Reply:
think because there's a few `&self` functions below and wanted interior mutability?

Reply:
yes, would it be preferred to change the functions to `&mut self` ?

Reply:
let's go with it as-is and see if it ends up being a performance bottleneck, don't expect it to matter a ton